borsh-derive/src/lib.rs

```diff
@@ -53,6 +53,29 @@ pub fn borsh_deserialize(input: TokenStream) -> TokenStream {
     })
 }
 
+#[proc_macro_derive(BorshBorrowedDeserialize, attributes(borsh_skip, borsh_init))]
```

Comment:
Add tests for the macros as well, we need to be sure they generate valid code (unless Borsh doesn't include tests for the macros at all)